packages/shared/routes/setup/CustomNetwork.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang="typescript">
+    import { localize } from '@core/i18n'
+    import { getOfficialNetworkConfig, INode, updateClientOptions } from '@core/network'
+    import { updateActiveProfileSettings } from '@core/profile'
+    import { appRouter } from '@core/router'
+    import { OnboardingLayout, Text, Button, Spinner } from 'shared/components'
+    import { NodeConfigurationForm } from 'shared/components/molecules'
+
+    let nodeConfigurationForm: NodeConfigurationForm
+    let nodeUrl: string
+    let isBusy = false
+
+    function onBackClick(): void {
+        $appRouter.previous()
+    }
+    function onSuccess(_isNetworkSwitch: boolean, node: INode, _oldNodeUrl: string): void {
+        const networkConfig = getOfficialNetworkConfig(node?.network?.protocol, node?.network?.type)
+        networkConfig.nodes = [{ ...node, isPrimary: true }]
+        updateClientOptions(networkConfig)
+        updateActiveProfileSettings({ networkConfig })
+        $appRouter.next()
+    }
```

Comment:
Could we extract this into an action on the network module. Additionally we do not have an active profile so we only need to set these settings in the newProfile store.